scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -764,8 +764,28 @@ class Router(formatOps: FormatOps) {
                 case b: Term.Block => b.tokens.head
                 case _ => assign.tokens.find(_.is[T.Equals]).get
               }
+              val assignIndent =
```

Comment:
this block is useful but unrelated to RHS optimal token, i am splitting off. it actually turned out to be a can of worms, somewhat similar to #1600 in size.